aware-phone/src/main/java/com/aware/phone/ui/Aware_Light_Client.java

```diff
@@ -469,8 +476,71 @@ protected void onPause() {
         if (prefs != null) prefs.unregisterOnSharedPreferenceChangeListener(this);
     }
 
+    public static boolean isBatteryOptimizationIgnored(Context context, String package_name) {
+        boolean is_ignored = true;
+        if (Build.VERSION.SDK_INT > Build.VERSION_CODES.LOLLIPOP_MR1) {
+            PowerManager pm = (PowerManager) context.getApplicationContext().getSystemService(Context.POWER_SERVICE);
+            is_ignored = pm.isIgnoringBatteryOptimizations(package_name);
+        }
+
+        if (!is_ignored) {
+            NotificationCompat.Builder mBuilder = new NotificationCompat.Builder(context, Aware.AWARE_NOTIFICATION_CHANNEL_GENERAL);
+            mBuilder.setSmallIcon(com.aware.R.drawable.ic_stat_aware_recharge);
+            mBuilder.setContentTitle(context.getApplicationContext().getResources().getString(com.aware.R.string.aware_activate_battery_optimize_ignore_title));
+            mBuilder.setContentText(context.getApplicationContext().getResources().getString(com.aware.R.string.aware_activate_battery_optimize_ignore));
+            mBuilder.setAutoCancel(true);
+            mBuilder.setOnlyAlertOnce(true); //notify the user only once
+            mBuilder.setDefaults(NotificationCompat.DEFAULT_ALL);
+            mBuilder = setNotificationProperties(mBuilder, AWARE_NOTIFICATION_IMPORTANCE_GENERAL);
+
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O)
+                mBuilder.setChannelId(Aware.AWARE_NOTIFICATION_CHANNEL_GENERAL);
+
+            Intent batteryIntent = new Intent(Settings.ACTION_IGNORE_BATTERY_OPTIMIZATION_SETTINGS);
+            batteryIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TOP);
+
+            PendingIntent clickIntent = PendingIntent.getActivity(context, 0, batteryIntent, PendingIntent.FLAG_UPDATE_CURRENT);
+            mBuilder.setContentIntent(clickIntent);
+
+            NotificationManager notManager = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
+            notManager.notify(Aware.AWARE_BATTERY_OPTIMIZATION_ID, mBuilder.build());
+        }
+
+        Log.d(Aware.TAG, "Battery Optimizations: " + is_ignored);
+
+        return is_ignored;
+    }
+
+    @Override
+    protected void onStop() {
+        // Check if the activity is finishing
+        boolean isFinishing = this.isFinishing();
+
+        // Handle based on whether it's system-initiated closure
+        if (!isFinishing) {
+            if (isBatteryOptimizationIgnored(this, "com.aware.phone")) {
+                Log.d("AWARE-Light_Client", "AWARE-Light stopped from background: may caused by battery optimization");
```

Comment:
There are three more instances here that could also be changed to "may be caused by", thanks!

Reply:
Changes made!